scipy/stats/tests/test_continuous.py

```diff
@@ -1109,3 +1110,100 @@ def test_ContinuousDistribution__str__(self):
+        with pytest.raises(ValueError, match=message):
+            Mixture([Normal()], weights=[0.5, 0.5])
+
+        message = "`weights` must have floating point dtype."
+        with pytest.raises(ValueError, match=message):
+            Mixture([Normal()], weights=[1])
+
+        message = "`weights` must have floating point dtype."
+        with pytest.raises(ValueError, match=message):
+            Mixture([Normal()], weights=[1])
+
+        message = "`weights` must sum to 1.0."
+        with pytest.raises(ValueError, match=message):
+            Mixture([Normal(), Normal()], weights=[0.5, 1.0])
+
+        message = "All `weights` must be non-negative."
+        with pytest.raises(ValueError, match=message):
+            Mixture([Normal(), Normal()], weights=[1.5, -0.5])
+
+    def test_basic(self):
+        rng = np.random.default_rng(582348972387243524)
+        X = Mixture((Normal(mu=-0.25, sigma=1.1), Normal(mu=0.5, sigma=0.9)),
+                    weights=(0.4, 0.6))
+        Y = MixedDist()
+        x = rng.random(10)
+
+        assert_allclose(X.logentropy(), Y.logentropy())
+        assert_allclose(X.entropy(), Y.entropy())
+        assert_allclose(X.mode(), Y.mode())
+        assert_allclose(X.median(), Y.median())
+        assert_allclose(X.mean(), Y.mean())
+        assert_allclose(X.variance(), Y.variance())
+        assert_allclose(X.standard_deviation(), Y.standard_deviation())
+        assert_allclose(X.skewness(), Y.skewness())
+        assert_allclose(X.kurtosis(), Y.kurtosis())
+        assert_allclose(X.logpdf(x), Y.logpdf(x))
+        assert_allclose(X.pdf(x), Y.pdf(x))
+        assert_allclose(X.logcdf(x), Y.logcdf(x))
+        assert_allclose(X.cdf(x), Y.cdf(x))
+        assert_allclose(X.logccdf(x), Y.logccdf(x))
+        assert_allclose(X.ccdf(x), Y.ccdf(x))
+        assert_allclose(X.ilogcdf(x), Y.ilogcdf(x))
+        assert_allclose(X.icdf(x), Y.icdf(x))
+        assert_allclose(X.ilogccdf(x), Y.ilogccdf(x))
+        assert_allclose(X.iccdf(x), Y.iccdf(x))
+        for kind in ['raw', 'central', 'standardized']:
+            for order in range(5):
+                assert_allclose(X.moment(order, kind=kind),
+                                Y.moment(order, kind=kind),
+                                atol=1e-15)
+
+        # weak test of `sample`
+        shape = (10, 20, 5)
+        y = X.sample(shape, rng=rng)
+        assert y.shape == shape
+        assert stats.ks_1samp(y.ravel(), X.cdf).pvalue > 0.05
+
+    def test_properties(self):
+        components = [Normal(mu=-0.25, sigma=1.1), Normal(mu=0.5, sigma=0.9)]
+        weights = (0.4, 0.6)
+        X = Mixture(components, weights=weights)
+
+        # Replacing properties doesn't work
+        with pytest.raises(AttributeError, match="...'Mixture' object has no setter"):
+            X.components = 10
+        with pytest.raises(AttributeError, match="...'Mixture' object has no setter"):
+            X.weights = 10
```

Comment:
Should have seen that coming...
```suggestion
        # Different version of Python have different messages
        with pytest.raises(AttributeError):
            X.components = 10
        with pytest.raises(AttributeError):
            X.weights = 10
```

Reply:
We may want to write a custom `fset` that raises with a standard error message after all, but yeah, for now I think it's fine to just not test the message.